test/fixtures/detect-existing-browser-api-object/content.js

```diff
@@ -10,12 +10,27 @@ test("browser api object in content script", (t) => {
     // On Firefox, window is not the global object for content scripts, and so we expect window.browser to not
     // be defined.
     t.equal(window.browser, undefined, "window.browser is expected to be undefined on Firefox");
+
+    try {
+      await browser.storage.sync.set({a: 'a'.repeat(10000000)});
+      t.fail('It should throw when attempting to set an object over quota');
+    } catch (error) {
+      t.ok(error instanceof Error);
+    }
   } else {
     // Check that the polyfill has created a wrapped API namespace as expected.
     t.notEqual(browser.runtime, chrome.runtime, "browser.runtime and chrome.runtime should not be equal");
     // On chrome, window is the global object and so the polyfilled browser API should
     // be also equal to window.browser.
     t.equal(browser, window.browser, "browser and window.browser should be the same object");
+
+    chrome.storage.local.set({a: 'a'.repeat(10000000)}, () => {
+      t.ok(chrome.runtime.lastError, 'It should throw when attempting to set an object over quota');
+      t.equal(chrome.runtime.lastError.message, 'QUOTA_BYTES quota exceeded');
+      t.notOk(chrome.runtime.lastError instanceof Error);
+    });
+
+    t.end();
```

Comment:
Does this `t.end();` call cause the test runner to pause until the callback for `chrome.storage.local.set` has been called?

Reply:
I used t.plan in https://github.com/mozilla/webextension-polyfill/commit/f6aece9aea78ff4d1b9a7f9d41d6f8ef54fc5ff0 to ensure the number of assertions is the same for both browsers